tests/test_xmltodict.py

```diff
@@ -295,3 +297,76 @@ def force_list(path, key, value):
             },
         }
         self.assertEqual(parse(xml, force_list=force_list, dict_constructor=dict), expectedResult)
+
+    def test_disable_entities_true_ignores_xmlbomb(self):
+        xml = """
+        <!DOCTYPE xmlbomb [
+            <!ENTITY a "1234567890" >
+            <!ENTITY b "&a;&a;&a;&a;&a;&a;&a;&a;">
+            <!ENTITY c "&b;&b;&b;&b;&b;&b;&b;&b;">
+        ]>
+        <bomb>&c;</bomb>
+        """
+        expectedResult = {'bomb': None}
+        try:
+            parse_attempt = parse(xml, disable_entities=True)
+        except expat.ExpatError:
+            self.assertTrue(True)
+        else:
+            self.assertEqual(parse_attempt, expectedResult)
+
+    def test_disable_entities_false_returns_xmlbomb(self):
+        xml = """
+        <!DOCTYPE xmlbomb [
+            <!ENTITY a "1234567890" >
+            <!ENTITY b "&a;&a;&a;&a;&a;&a;&a;&a;">
+            <!ENTITY c "&b;&b;&b;&b;&b;&b;&b;&b;">
+        ]>
+        <bomb>&c;</bomb>
+        """
+        bomb = "1234567890" * 64
+        expectedResult = {'bomb': bomb}
+        self.assertEqual(parse(xml, disable_entities=False), expectedResult)
+
+    def test_disable_entities_true_ignores_external_dtd(self):
+        xml = """
+        <!DOCTYPE external [
+            <!ENTITY ee SYSTEM "http://www.python.org/">
+        ]>
+        <root>&ee;</root>
+        """
+        expectedResult = {'root': None}
+        try:
+            parse_attempt = parse(xml, disable_entities=True)
+        except expat.ExpatError:
+            self.assertTrue(True)
+        else:
+            self.assertEqual(parse_attempt, expectedResult)
+
+    def test_disable_entities_true_attempts_external_dtd(self):
+        xml = """
+        <!DOCTYPE external [
+            <!ENTITY ee SYSTEM "http://www.python.org/">
+        ]>
+        <root>&ee;</root>
+        """
+        def raising_external_ref_handler(*args, **kwargs):
+            parser = ParserCreate(*args, **kwargs)
+            parser.ExternalEntityRefHandler = lambda *x: 0
+            try:
+                feature = "http://apache.org/xml/features/disallow-doctype-decl"
+                parser._reader.setFeature(feature, True)
+            except AttributeError:
+                pass
+            return parser
+        expat.ParserCreate = raising_external_ref_handler
+        # Using this try/catch because a TypeError is thrown before 
+        # the ExpatError, and Python 2.6 is confused by that.
+        try:
+            parse(xml, disable_entities=False, expat=expat)
+        except expat.ExpatError:
+            self.assertTrue(True)
+        else:
+            self.assertTrue(False)
```

Comment:
Thanks for having a look at this :smile:

The comment in the lines just above it is (not very clearly) trying to say that the reason for this ugly try/except block test is that testing with assertRaises as suggested doesn't work with Python 2.6. 

In this case, a TypeError gets thrown before the ExpatError of interest, and Py2.6's assertRaises gets confused by that (considers the test a failure, I think), whereas other Python versions and implemenations listed in tox don't have this problem.

I think I tried using a tuple of (TypeError, expat.ExpatError) with assertRaises and also nesting two assertRaises context managers, but Py2.6 was having none of it. The code before you worked though.

If it'd help I could elaborate on the existing comment and add it to the PR?